mono/metadata/threads.c

```diff
@@ -2194,8 +2194,14 @@ void mono_thread_current_check_pending_interrupt ()
 void
 ves_icall_System_Threading_Thread_Abort (MonoInternalThread *thread, MonoObject *state)
 {
+	MonoThread *sys_thread = mono_thread_current ();
+
 	LOCK_THREAD (thread);
 	
+	if (sys_thread->pending_exception) {
```

Comment:
Yea, this block is not necessary. (I'm pretty sure it's a no-op, actually).  Sorry about that.

Your test doesn't actually exercise the changed behavior because the throw of "PendingException" above just unwinds starting from managed - the exception isn't pending, it's just thrown. (You'd need an icall to post a pending exception instead of the throw)
When the ordinary exception is caught, the JIT will continue throwing the Abort when leaving the catch clause for the ordinary exception.

We have `thread6.cs` and `threadpool-exception5.cs` that exercise the interaction between ordinary exceptions and abort.